plugins/catalog-backend/src/ingestion/reader.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Location } from '../catalog';
+import { ParserOutput } from '../descriptors';
+import { FileLocationReader, LocationReader } from './model';
+
+// Registry of all readers
+const readers: LocationReader[] = [new FileLocationReader()];
```

Comment:
Probably worth making this an object tbh, with key being the type. Or a Map.

Reply:
Why? :) I made it the same as the other one (that's multi-keyed) just shrugging at it.

Reply:
It ensures uniqueness without any extra work

Reply:
But it's a hard-coded manual list. And any duplicates would be a fatal bug anyway :) hiding it won't make it better. Then we'd be better of making code for runtime checks of we're at all worried about that...

Reply:
I mean the outcome is essentially the same. Yours: last registered wins. Mine: first registered wins :)

Reply:
No I mean an object literal here, also helps with discoverability, finding i.e. finding the reader for a certain time.

Something like

```ts
const readers = {
  file: new FileLocationReader(),
}
```

Reply:
Aha but I wanted to localise the knowledge of what it matches in one place :) see the version/kind stuff for a parallel